test/test_api.py

```diff
@@ -284,6 +284,17 @@ def test06b_resources_direction(self, mocker):
         self.assertTrue(get_uri(args).endswith('/resources/image/tags/{}'.format(API_TEST_TAG)))
         self.assertEqual(get_params(args)['direction'], 'desc')
 
+    @patch('urllib3.request.RequestMethods.request')
+    @unittest.skipUnless(cloudinary.config().api_secret, "requires api_key/api_secret")
+    def test_extra_headers(self, mocker):
+        """Should support extra headers"""
+        mocker.return_value = MOCK_RESPONSE
+        uploader.upload(TEST_IMAGE, extra_headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) '
+                                                                 'AppleWebKit/537.36 (KHTML, like Gecko) '
+                                                                 'Chrome/58.0.3029.110 Safari/537.3'})
+        headers = get_headers(mocker.call_args[0])
+        self.assertTrue(headers.get('User-Agent'))
```

Comment:
it is better to assert the actual value, using `assertEqual`. 
`assertTrue` just checks whether the string is not empty, like, if you remove `extra_headers` parameter from the function call, the test will still pass.